src/charm.py

```diff
@@ -206,14 +206,6 @@ def _on_database_endpoints_changed(self, event: DatabaseEndpointsChangedEvent) -
             fd.write(self._connection_string)
             os.fsync(fd)
 
-        try:
-            os.kill(int(self.app_peer_data[PROC_PID_KEY]), signal.SIGKILL)
-        except ProcessLookupError:
-            del self.app_peer_data[PROC_PID_KEY]
-            return
-        count = self._count_writes()
-        self._start_continuous_writes(count + 1)
-
```

Comment:
If the connection will not block, we don't have to restart the script on db switchover